### PR TITLE
lzham_codec: add package

### PR DESCRIPTION
**Pull request by Chi-EEE**

### CODE REVIEW

File: packages/l/lzham_codec/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+    set_urls("https://github.com/richgel999/lzham_codec.git")
+
+    add_versions("2023.01.03", "d379b1f9121e2197881c61cfc4713c78848bdfe7")
+
+    on_install(function (package)
+        os.cp(path.join(package:scriptdir(), "port", "xmake.lua"), "xmake.lua")
+        io.replace("lzhamdecomp/lzham_symbol_codec.h", [[#include "lzham_prefix_coding.h"]], [[#include "lzham_prefix_coding.h"
+#include <cstdint>]], {plain = true})
+        import("package.tools.xmake").install(package, {kind = package:config("shared") and "shared" or "static"})
```
Comment (waruqi) [inline packages/l/lzham_codec/xmake.lua:13]: tools.xmake will pass kind automatically, we need not add it. you can remove it. `{kind = package:config("shared") and "shared" or "static"}`

File: packages/l/lzham_codec/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        assert(package:check_cxxsnippets({test = [[
+            #include <lzham_static_lib.h>
+            void test() {
+                lzham_static_lib lzham_lib;
+                lzham_lib.load();
+            }
+        ]]}))
+    end)
+
+package_end()
```
Comment (waruqi) [inline packages/l/lzham_codec/xmake.lua:29]: remove it.

**Comment by waruqi** [comment]
`#include "string.h"`  is missing for mingw

**Comment by waruqi** [comment]
there is not malloc_usable_size in mingw. maybe need patch code.

**Comment by Chi-EEE** [comment]
Somehow the USE_ASSERT_MIMALLOC define is not being used in test but if I test locally with is-plat("windows"), the mimalloc define would be used

**Comment by waruqi** [comment]
> 



> Somehow the USE_ASSERT_MIMALLOC define is not being used in test but if I test locally with is_plat("windows"), the mimalloc define would get used

you need add `add_defines("USE_ASSERT_MIMALLOC")` in package scope.

File: packages/l/lzham_codec/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            package:add("deps", "mimalloc")
+        end
+    end)
+
+    on_install(function (package)
+        os.cp(path.join(package:scriptdir(), "port", "xmake.lua"), "xmake.lua")
+        if package:config("mimalloc") then
+            package:add("defines", "ASSERT_USE_MIMALLOC")
+        end
+        import("package.tools.xmake").install(package)
```
Comment (waruqi) [inline packages/l/lzham_codec/xmake.lua:28]: you need pasd mimalloc config to inner option.

loca configs = {}
configs.mimalloc = package:config("mimalloc")
import("package.tools.xmake").install(package, configs)

File: packages/l/lzham_codec/port/xmake.lua [hunk 4]
```diff
@@ -0,0 +1,107 @@
+add_rules("mode.debug", "mode.release")
+
+if has_config("mimalloc") then
```
Comment (waruqi) [inline packages/l/lzham_codec/port/xmake.lua:3]: you need define a option.

```lua
option("mimalloc")
```

**Pull request merged**